rust/tools/authorization-logic/raksha_examples/policy_patterns/NonUnanimousDecisions

```diff
@@ -0,0 +1,132 @@
+// determine if another principal can occupy a particular role. As in
+// role-based access control, whether or not a principal occupies a role
+// can be used to make access decisions. In this example the role is
+// "DeviceOwner". Other roles are also possible, and it is also possible
+// to make membership of a role conditional on other facts.
+"RoleAuthority" says {
+    "DataOwner2" canActAs "DeviceOwner".
+}
+
+// This is a policy that collects the preferences of both data owners
+// on whether or not they are OK with data being sent. However, the 
+// actual decision depends only on the preference of the principal
+// in a specifc role (DeviceOwner).
+
+// This role-based decision policy is one way of resolving the conflict, but 
+// the pattern can be re-used by changing the code in the decision policy
+// principal. For example, voting could also be implemented (if this language
+// were extended with aggregation).
+"RoleBasedDecisionPolicy" says {
+    // The "RoleAuthority" is trusted by the policy to decide who occupies
+    // the device owner role.
+    "RoleAuthority" canSay PrincipalX canActAs "DeviceOwner" :-
+        isPrincipal(PrincipalX).
+
+    // If there is a handle that depends on data from BOTH "datum1" and
+    // "datum2", in order to permit the particle to "send_remotely"
+    // (i.e. pass the check generated by "will("send_remotely", handleX),
+    // we need to prove BOTH `"DO1" says may("send_remotely", "datum1")`
+    // AND `"DO2" says may("send_remotely", "datum2"). Because both owners
+    // have delegated the right to make this decision to this policy, both
+    // owners will respect the outcome from the policy.
+  
+    // In this case, the preference of the device owner is the only one
+    // that matters for both pieces of data. (Here, it is assumed that there
+    // is just one device owner in the system).
+    "ExampleParticle" may("send_remotely", "datum1") :-
+        PrincipalX canActAs "DeviceOwner",
+        PrincipalX hasPreference("do_send_remotely").
+    "ExampleParticle" may("send_remotely", "datum2") :-
+        PrincipalX canActAs "DeviceOwner",
+        PrincipalX hasPreference("do_send_remotely").
+
+
+    // This just allows the policy to "import" the preferences of the two
+    // data owners. It means that the policy believes the owners about their
+    // own preferences.
+    PrincipalX canSay PrincipalX hasPreference(PrefX) :-
+        isPrincipal(PrincipalX), isPreference(PrefX).
+    
+    // This is just boilerplate that populates the universes
+    isPrincipal("DataOwner1").
+    isPrincipal("DataOwner2").
+    isPrincipal("RoleAuthority").
+    isPreference("do_send_remotely").
+    isPreference("do_not_send_remotely").
+}
+
+// These are the checks that would be generated by a particle
+// called "ExampleParticle" with "will("send_remotely", "handleX")"
+// and also hasTag("handleX", "datum1"), hasTag("handleX", "datum2").
```

Comment:
Hold on, this comment says "will" but the recipe body says "may". Is this a disconnect or is this intentional?

Reply:
This is intentional. This is meant to be an instance of the check described in the comment of [VisionPipelineAuthLogic](https://github.com/google-research/raksha/pull/170/files), which is repeated here:

```
// For each particle P, for each tag t in union over tags in read edges of P,
//      for each statement of the form:
//          `P says will <RELATION> using <TAG>`
//      prove
//          `ownsTag(Q, <TAG>)`
//          `Q says may <RELATION> using <HANDLE>`
```

The idea is that:
- data owners describe permitted us of data with `may`
- app specifiers describe usage that will happen on handles with `will`
- When the taint analysis says that some tags show up on a handle, we check that the intended use aligns with the permitted use (by using the "will"s to figure out some rules that check the "mays" that we generate)